packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -131,9 +132,7 @@ class MessagePayload {
       }
     }
 
-    const components = this.options.components?.map(c =>
-      isJSONEncodable(c) ? c.toJSON() : this.target.client.options.jsonTransformer(c),
-    );
+    const components = this.options.components?.map(c => new ActionRowBuilder(c).toJSON());
```

Comment:
if `c` is already an instance of `ActionRowBuilder` this will result in a nested `data` property of `{ type: 1, data: { type: 1 } }` since the constructor isn't designed to take class instances. To fix this you could either do an `instanceof` check or revert to the old behaviour and adjust the typings. It's a question of whether mixing builders and raw action row objects should be supported, if someone wanted to mix them they could always manually call `toJSON` on the component builders. An advantage of `isJSONEncodable` over `instanceof` is that it accepts a wider range of input, e.g. if you `import` a `ComponentBuilder` via esm from the builders package it wouldn't pass the `instanceof`-check since d.js uses the commonjs imports